pkg/k8s/watchers/node.go

```diff
@@ -70,12 +73,16 @@ func (k *K8sWatcher) nodesInit(k8sClient kubernetes.Interface) {
 }
 
 func (k *K8sWatcher) updateK8sNodeV1(oldK8sNode, newK8sNode *slim_corev1.Node) error {
-	oldNodeLabels := oldK8sNode.GetLabels()
+	var oldNodeLabels map[string]string
+	if oldK8sNode != nil {
+		oldNodeLabels = oldK8sNode.GetLabels()
+	}
 	newNodeLabels := newK8sNode.GetLabels()
 
 	nodeEP := k.endpointManager.GetHostEndpoint()
 	if nodeEP == nil {
-		log.Error("Host endpoint not found")
+		log.Debug("Host endpoint not found, updating node labels")
+		node.UpdateLabels(oldNodeLabels, newNodeLabels)
```

Comment:
Why not simply pass the newNodeLabels with the `node.SetLabels(newNodeLabels)`?

Reply:
:facepalm: I think I started implementing `UpdateLabels` thinking something different was needed and completely forgot about `SetLabels` afterward.